src/indicators/drought/arc2_precipitation.py

```diff
@@ -0,0 +1,884 @@
+                "first download using `download_data()`.",
+                raw_filepath.name,
+            )
+
+        return da
+
+    def download_data(
+        self, main: bool = False, replace_missing: Union[bool, str] = False
+    ):
+        """
+        Download ARC2 data for all dates between `self.date_min`
+        and `self.date_max`. If `main`, then all data
+        is downloaded directly from the servers and set as the
+        main file. If not `main`, then data
+        is only downloaded for dates not already
+        available in the main file, and then merged
+        into the main file.
+
+        :param main: Boolean on whether to set download as
+            main if `True`, or only download missing data
+            and merge to main if `False`.
+        :param replace_missing: Only relevant if not `main`.
+            If `True`, looks for missing values through entire
+            dataset and ensures they are re-downloaded. If an
+            ISO 8601 string is passed, then missing values are
+            only looked for on or after that date.
+        """
+        if main:
+            self._download(self.date_min, self.date_max, main)
+        else:
+            self._check_main_exists()
+            # load main data and find all dates covered
+            # compare to min/max and then download missing
+            mr = self.load_raw_data()
+
+            # drop missing data if requested
+            if replace_missing:
+                if isinstance(replace_missing, str):
+                    replace_date = datetime.fromisoformat(replace_missing)
+                else:
+                    replace_date = datetime.combine(
+                        self.date_min, datetime.min.time()
+                    )
+
+                t_subset = mr.indexes["T"] < replace_date
+                val_subset = np.max(
+                    np.max(mr.values != -999, array=1), array=1
+                )
+
+                # keeps rows if date is less than specified or
+                # value is different than -999 (missing)
+                mr = mr.loc[t_subset | val_subset, :, :]
+
+            # subtracting 12 hours to ensure they match with dates
+            # generated from pd.date_range()
+            loaded_dates = mr.indexes[
+                "T"
+            ].to_datetimeindex() - pd.to_timedelta("12:00:00")
+            full_dates = pd.date_range(self.date_min, self.date_max)
+            needed_dates = full_dates.difference(loaded_dates)
+
+            if len(needed_dates) > 0:
+                date_ranges = self._group_date_ranges(needed_dates)
+                for dates in date_ranges:
+                    self._download(
+                        date_min_dl=dates[0],
+                        date_max_dl=dates[1],
+                        main=False,
+                    )
+
+                self._sort_raw_data()
+            else:
+                logger.info("No additional data needs downloading.")
+
```

Comment:
Gonna drop `main` and move to the `clobber` system for much more simplicity. Will keep this here though for that refactor, because a nice and simple way to approach looking for missing dates and replacing them.